packages/wells/src/client/api/wellsApi.ts

```diff
@@ -131,12 +142,33 @@ export class WellsAPI extends ConfigureAPI {
       userFilter.polygon = undefined;
       filter = this.convertToApiWellFilter(userFilter);
     }
-    const path = this.getPath('/wells/list', cursor)
-    return await this.client?.asyncPost<WellItems>(path, {'data': filter})
-      .then(response => this.addLazyToAllWellItems(response.data))
-      .catch(err => {
-        throw new HttpError(err.status, err.errorMessage, {})
-    });
+    return filter
+  }
+
+  public filterSlow = async (userFilter: WellFilter): Promise<Well[] | undefined> => {
+    let wells: Well[] = []
+    let cursor = undefined
+    const filter: WellFilterAPI = this.wrapFilter(userFilter);
+    do {
+        const wellsChunk: WellItems | undefined = await this.sendFilterRequest(filter, cursor);
+        if (wellsChunk == undefined) {
+          break
+        }
+
+        if (wellsChunk.items.length > 0) {
+          wells = wells.concat(wellsChunk.items)
+        }
```

Comment:
why check the length? Can't you just write `wells = wells.concat(wellsChunk.item)`?